### PR TITLE
Add Async Batch Shard Settings

### LINKED ISSUE #7055: [DOC] Update documentation for cluster settings to include Async shard batch fetch settings.
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
Recent optimisations were made in the Async Shard Fetch flow to now fetch in batches of Shards on a node level basis instead of being executed per shard per node. The documentation changes required to be updated include settings to - 
1. Enable Async Batch Fetch on existing shard allocator 
2. Set Batch sizes for the Gateway allocators

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
https://github.com/opensearch-project/OpenSearch/issues/5098
https://github.com/opensearch-project/OpenSearch/pull/8746

### PR DESCRIPTION
**Pull request by Pranshu-S**
### Description
Adds documentation for enabling Async Batch Fetch on a cluster and setting Batch sizes to the gateway allocators as per the changes/optimisations done on  Async Shard Fetch flow
 
### Issues Resolved
Resolves #7055 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
Thanks, @Pranshu-S ! Are these changes being released in 2.14 or should they be backported to 2.13?

**Comment by Pranshu-S** [comment]
> Thanks, @Pranshu-S ! Are these changes being released in 2.14 or should they be backported to 2.13?

This will be for 2.14.

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 1]
```diff
@@ -96,6 +96,8 @@ OpenSearch supports the following cluster-level routing and shard allocation set
     - `REPLICA_FIRST` – Replica shards are relocated first, before primary shards. This prioritization may help prevent a cluster's health status from going red when carrying out shard relocation in a mixed-version, segment-replication-enabled OpenSearch cluster. In this situation, primary shards relocated to OpenSearch nodes of a newer version could try to copy segment files to replica shards on an older version of OpenSearch, which would result in shard failure. Relocating replica shards first may help to avoid this in multi-version clusters. 
     - `NO_PREFERENCE` – The default behavior in which the order of shard relocation has no importance. 
 
+- `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards we send in one batch to data nodes for fetching metadata. Default is `2000`.
+- `cluster.allocator.existing_shards_allocator.batch_enabled` (Boolean): Boolean setting to enable/disable batch allocation of unassigned shards already existing on disk. This will allow sending all Unassigned Shards to the ExistingShard Allocator to make decision to allocate in one or more go. Default is `False`.
```
Comment (Naarcha-AWS) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:100]:
```suggestion
- `cluster.allocator.existing_shards_allocator.batch_enabled` (Boolean): Enables batch allocation of unassigned shards already existing on the disk. This allows for sending all `Unassigned`shards to the `ExistingShard` allocator. The allocator decides how to allocate the unassigned shards. Default is `False`.
```

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 2]
```diff
@@ -96,6 +96,8 @@ OpenSearch supports the following cluster-level routing and shard allocation set
     - `REPLICA_FIRST` – Replica shards are relocated first, before primary shards. This prioritization may help prevent a cluster's health status from going red when carrying out shard relocation in a mixed-version, segment-replication-enabled OpenSearch cluster. In this situation, primary shards relocated to OpenSearch nodes of a newer version could try to copy segment files to replica shards on an older version of OpenSearch, which would result in shard failure. Relocating replica shards first may help to avoid this in multi-version clusters. 
     - `NO_PREFERENCE` – The default behavior in which the order of shard relocation has no importance. 
 
+- `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards we send in one batch to data nodes for fetching metadata. Default is `2000`.
```
Comment (Naarcha-AWS) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:99]:
```suggestion
- `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards sent in one batch to data nodes for fetching metadata. Default is `2000`.
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: _install-and-configure/configuring-opensearch/cluster-settings.md (hunk 1 above)
Comment (Pranshu-S) [reply _install-and-configure/configuring-opensearch/cluster-settings.md]: Addressed changes in next commit. Thanks!

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 3]
```diff
@@ -96,6 +96,8 @@ OpenSearch supports the following cluster-level routing and shard allocation set
     - `REPLICA_FIRST` – Replica shards are relocated first, before primary shards. This prioritization may help prevent a cluster's health status from going red when carrying out shard relocation in a mixed-version, segment-replication-enabled OpenSearch cluster. In this situation, primary shards relocated to OpenSearch nodes of a newer version could try to copy segment files to replica shards on an older version of OpenSearch, which would result in shard failure. Relocating replica shards first may help to avoid this in multi-version clusters. 
     - `NO_PREFERENCE` – The default behavior in which the order of shard relocation has no importance. 
 
+- `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards sent in one batch to data nodes for fetching metadata. Default is `2000`.
```
Comment (shwetathareja) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:99]: can we change it to -> Limits the number of shards sent in one batch to data nodes for **fetching unassigned shard** metadata

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 4]
```diff
@@ -96,6 +96,8 @@ OpenSearch supports the following cluster-level routing and shard allocation set
...
+- `cluster.allocator.existing_shards_allocator.batch_enabled` (Boolean): Enables batch allocation of unassigned shards already existing on the disk. This allows for sending all `Unassigned`shards to the `ExistingShard` allocator. The allocator decides how to allocate the unassigned shards. Default is `False`.
```
Comment (shwetathareja) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:100]: Enables batch allocation of unassigned shards already existing on the disk as opposed to one shard at a time. This reduces the memory and transport overhead by fetching unassigned shard metadata in a batch call. Default is `False`.

Comment (Pranshu-S) [reply _install-and-configure/configuring-opensearch/cluster-settings.md]: Updated the wordings.

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 5]
```diff
@@ -96,6 +96,8 @@ OpenSearch supports the following cluster-level routing and shard allocation set
     - `REPLICA_FIRST` – Replica shards are relocated first, before primary shards. This prioritization may help prevent a cluster's health status from going red when carrying out shard relocation in a mixed-version, segment-replication-enabled OpenSearch cluster. In this situation, primary shards relocated to OpenSearch nodes of a newer version could try to copy segment files to replica shards on an older version of OpenSearch, which would result in shard failure. Relocating replica shards first may help to avoid this in multi-version clusters. 
     - `NO_PREFERENCE` – The default behavior in which the order of shard relocation has no importance. 
 
+- `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards sent in one batch to data nodes for fetching unassigned shard metadata. Default is `2000`.
+- `cluster.allocator.existing_shards_allocator.batch_enabled` (Boolean): Enables batch allocation of unassigned shards already existing on the disk as opposed to one shard at a time. This reduces the memory and transport overhead by fetching unassigned shard metadata in a batch call. Default is False.
```
Comment (Naarcha-AWS) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:100]:
```suggestion
- `cluster.allocator.existing_shards_allocator.batch_enabled` (Boolean): Enables batch allocation of unassigned shards which already exist on the disk as opposed to one shard at a time. This reduces the memory and transport overhead by fetching any unassigned shard metadata in a batch call. Default is `False`.
```

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 6]
```diff
@@ -96,6 +96,8 @@ OpenSearch supports the following cluster-level routing and shard allocation set
     - `REPLICA_FIRST` – Replica shards are relocated first, before primary shards. This prioritization may help prevent a cluster's health status from going red when carrying out shard relocation in a mixed-version, segment-replication-enabled OpenSearch cluster. In this situation, primary shards relocated to OpenSearch nodes of a newer version could try to copy segment files to replica shards on an older version of OpenSearch, which would result in shard failure. Relocating replica shards first may help to avoid this in multi-version clusters. 
     - `NO_PREFERENCE` – The default behavior in which the order of shard relocation has no importance. 
 
+- `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards sent in one batch to data nodes for fetching unassigned shard metadata. Default is `2000`.
```
Comment (Naarcha-AWS) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:99]:
```suggestion
- `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards sent to data nodes in one batch meant for fetching unassigned shard metadata. Default is `2000`.
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]
A couple more small tweaks.

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 7]
```diff
@@ -96,6 +96,8 @@ OpenSearch supports the following cluster-level routing and shard allocation set
     - `REPLICA_FIRST` – Replica shards are relocated first, before primary shards. This prioritization may help prevent a cluster's health status from going red when carrying out shard relocation in a mixed-version, segment-replication-enabled OpenSearch cluster. In this situation, primary shards relocated to OpenSearch nodes of a newer version could try to copy segment files to replica shards on an older version of OpenSearch, which would result in shard failure. Relocating replica shards first may help to avoid this in multi-version clusters. 
     - `NO_PREFERENCE` – The default behavior in which the order of shard relocation has no importance. 
 
+- `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards sent to data nodes in one batch meant for fetching unassigned shard metadata. Default is `2000`.
```
Comment (Naarcha-AWS) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:99]:
```suggestion
- `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards sent to data nodes in one batch for fetching any unassigned shard metadata. Default is `2000`.
```

**Review by Naarcha-AWS — APPROVED** [review]

File: _install-and-configure/configuring-opensearch/cluster-settings.md [hunk 8]
```diff
@@ -96,6 +96,8 @@ OpenSearch supports the following cluster-level routing and shard allocation set
     - `REPLICA_FIRST` – Replica shards are relocated first, before primary shards. This prioritization may help prevent a cluster's health status from going red when carrying out shard relocation in a mixed-version, segment-replication-enabled OpenSearch cluster. In this situation, primary shards relocated to OpenSearch nodes of a newer version could try to copy segment files to replica shards on an older version of OpenSearch, which would result in shard failure. Relocating replica shards first may help to avoid this in multi-version clusters. 
     - `NO_PREFERENCE` – The default behavior in which the order of shard relocation has no importance. 
 
+- `cluster.allocator.gateway.batch_size` (Integer): Limits the number of shards sent to data nodes in one batch for fetching any unassigned shard metadata. Default is `2000`.
+- `cluster.allocator.existing_shards_allocator.batch_enabled` (Boolean): Enables batch allocation of unassigned shards which already exist on the disk as opposed to one shard at a time. This reduces the memory and transport overhead by fetching any unassigned shard metadata in a batch call. Default is `False`.
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:100]:
```suggestion
- `cluster.allocator.existing_shards_allocator.batch_enabled` (Boolean): Enables batch allocation of unassigned shards that already exist on the disk as opposed to allocating one shard at a time. This reduces memory and transport overhead by fetching any unassigned shard metadata in a batch call. Default is `False`.
```

Comment (natebower) [inline _install-and-configure/configuring-opensearch/cluster-settings.md:100]: Is `false` intentionally capitalized?

**Review by natebower — COMMENTED** [review]
@Pranshu-S @Naarcha-AWS Just a couple comments/changes. Thanks!

File: _install-and-configure/configuring-opensearch/cluster-settings.md (hunk 8 above)
Comment (Naarcha-AWS) [reply _install-and-configure/configuring-opensearch/cluster-settings.md:100]: Should be lower-cased?

**Pull request merged**